pkg/bgpv1/api/conversions.go

```diff
@@ -261,3 +261,180 @@ func ToAgentRoutes(ms []*models.BgpRoute) ([]*types.Route, error) {
+	var ret []*types.RoutePolicyStatement
+	var retErr error
+
+	for _, s := range statements {
+		stmt, err := ToAgentRoutePolicyStatement(s)
+		if err != nil {
+			retErr = errors.Join(retErr, err)
+			continue
+		}
+		ret = append(ret, stmt)
+	}
+	return ret, retErr
+}
+
+func ToAPIRoutePolicyStatement(s *types.RoutePolicyStatement) *models.BgpRoutePolicyStatement {
+	localPref := int64(-1)
+	if s.Actions.SetLocalPreference != nil {
+		localPref = *s.Actions.SetLocalPreference
+	}
+	ret := &models.BgpRoutePolicyStatement{
+		MatchNeighbors:      s.Conditions.MatchNeighbors,
+		MatchPrefixes:       ToApiMatchPrefixes(s.Conditions.MatchPrefixes),
+		RouteAction:         ToApiRoutePolicyAction(s.Actions.RouteAction),
+		AddCommunities:      s.Actions.AddCommunities,
+		AddLargeCommunities: s.Actions.AddLargeCommunities,
+		SetLocalPreference:  localPref,
+	}
+	return ret
+}
+
+func ToAgentRoutePolicyStatement(s *models.BgpRoutePolicyStatement) (*types.RoutePolicyStatement, error) {
+	var localPref *int64
+	if s.SetLocalPreference >= 0 {
+		localPref = &s.SetLocalPreference
+	}
+	prefixes, err := ToAgentMatchPrefixes(s.MatchPrefixes)
+	if err != nil {
+		return nil, err
+	}
+	ret := &types.RoutePolicyStatement{
+		Conditions: types.RoutePolicyConditions{
+			MatchNeighbors: s.MatchNeighbors,
+			MatchPrefixes:  prefixes,
+		},
+		Actions: types.RoutePolicyActions{
+			RouteAction:         ToAgentRoutePolicyAction(s.RouteAction),
+			AddCommunities:      s.AddCommunities,
+			AddLargeCommunities: s.AddLargeCommunities,
+			SetLocalPreference:  localPref,
+		},
+	}
+	return ret, nil
+}
+
+func ToApiMatchPrefixes(prefixes []*types.RoutePolicyPrefixMatch) []*models.BgpRoutePolicyPrefixMatch {
+	var ret []*models.BgpRoutePolicyPrefixMatch
+	for _, p := range prefixes {
+		ret = append(ret, &models.BgpRoutePolicyPrefixMatch{
+			Cidr:         p.CIDR.String(),
+			PrefixLenMin: int64(p.PrefixLenMin),
+			PrefixLenMax: int64(p.PrefixLenMax),
+		})
+	}
+	return ret
+}
+
+func ToAgentMatchPrefixes(prefixes []*models.BgpRoutePolicyPrefixMatch) ([]*types.RoutePolicyPrefixMatch, error) {
+	var ret []*types.RoutePolicyPrefixMatch
+	var retErr error
+
+	for _, p := range prefixes {
+		cidr, err := netip.ParsePrefix(p.Cidr)
+		if err != nil {
+			retErr = errors.Join(retErr, err)
+			continue
+		}
+		ret = append(ret, &types.RoutePolicyPrefixMatch{
+			CIDR:         cidr,
+			PrefixLenMin: int(p.PrefixLenMin),
+			PrefixLenMax: int(p.PrefixLenMax),
+		})
+	}
+	return ret, retErr
+}
+
+func ToApiRoutePolicyType(t types.RoutePolicyType) string {
+	if t == types.RoutePolicyTypeExport {
+		return "export"
+	}
+	return "import"
+}
+
+func ToAgentRoutePolicyType(t string) types.RoutePolicyType {
+	if t == "export" {
+		return types.RoutePolicyTypeExport
+	}
+	return types.RoutePolicyTypeImport
+}
+
+func ToApiRoutePolicyAction(a types.RoutePolicyAction) string {
+	switch a {
+	case types.RoutePolicyActionNone:
+		return "none"
```

Comment:
Since we are checking on these strings might make sense to make them constants?

Reply:
fixed (and the other comments as well) in https://github.com/cilium/cilium/pull/28973/commits/4360104ce89629213e1c9e77a9330f1bed2c7904